File: lib/models/equipment.dart
import '../models/peripherals.dart';

class Equipment {
  final String id;
  final String name;
  final String description;
  final String userId; // ID del usuario asignado
  final List<Peripheral> peripherals;
  final String qrCode;
  final String? serialNumber; // Número de serie del equipo (opcional)
  final DateTime assignmentDate;

  Equipment({
    required this.id,
    required this.name,
    required this.description,
    required this.userId,
    required this.peripherals,
    required this.qrCode,
    this.serialNumber,
    required this.assignmentDate,
  });

  Map<String, dynamic> toMap() {
    return {
      '_id': id,
      'name': name,
      'description': description,
      'userId': userId,
      'peripherals': peripherals.map((p) => p.toMap()).toList(),
      'qrCode': qrCode,
      'serialNumber': serialNumber,
      'assignmentDate': assignmentDate.toIso8601String(),
    };
  }

  factory Equipment.fromMap(Map<String, dynamic> map) {
    return Equipment(
      id: map['_id'],
      name: map['name'],
      description: map['description'],
      userId: map['userId'],
      peripherals: (map['peripherals'] as List)
          .map((p) => Peripheral.fromMap(p))
          .toList(),
      qrCode: map['qrCode'],
      serialNumber: map['serialNumber'],
      assignmentDate: DateTime.parse(map['assignmentDate']),
    );
  }
}

File: api/models/equipment.js
const mongoose = require('mongoose');

const equipmentSchema = new mongoose.Schema({
  qrCode: {
    type: String,
    required: true,
    unique: true
  },
  name: {
    type: String,
    required: true
  },
  description: {
    type: String,
    required: true
  },
  userId: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'User',
    required: true
  },
  peripherals: [{
    types: {
      type: Map,
      of: Boolean,
      default: {}
    },
    serialNumber: String,
    isAssigned: {
      type: Boolean,
      default: true
    }
  }],
  serialNumber: String,
  assignmentDate: {
    type: Date,
    default: Date.now
  }
});

module.exports = mongoose.model('Equipment', equipmentSchema);